### ISSUE TITLE
Intercept the span for user input

### ISSUE THREAD

**Issue by arichiardi** [replit/jq-console#79]
Hello @amasad a question...

Is there a way to intercept, modify, or customize what is the user input `<span>` ?

It is exactly between `jqconsole-prompt` and `jqconsole-cursor` and I would need to customize it in order to provide real time syntax coloring.

Is it difficult to achieve?

**Comment by arichiardi** [comment]
The best would be to be able to intercept the characters, parse them, and write out ANSI (which is supported in `Write` already).

Another idea is to inject my own `<pre><code>` or other `class` so that `highlightjs` can work on it. but I guess it will result in higher execution time & bad performance.

**Comment by arichiardi** [comment]
Hello @amasad, probably the feature I am looking for is there somewhere. From the README I see:

* jqconsole-input: The prompt area during input. May have multiple lines.
* jqconsole-old-input: Previously-entered inputs.

Is it exactly what is between the prompt and the cursor?

The problem is I don't see them in the DOM. Maybe I need to do something to enable it?

**Comment by arichiardi** [comment]
I saw that I basically need to replace `$prompt-left` but when I do:

```coffeescript
 # Sets the text prompt escaping the html and outputting ANSI
  #   @arg text: The text to append. Can contain multiple lines.
  #   @arg escape: Whether the text to write should be html escaped. Optional, defaults to true.
  _FillPromptText: (text, escape=true) ->
    console.log @$prompt_left
    @$prompt_left = @ansi.stylize $(EMPTY_SPAN).text(@$prompt_left.text() + text).html()
```

I get a big `Uncaught TypeError: this.$prompt_left.text is not a function`, maybe I am missing something here...

**Comment by arichiardi** [comment]
Oh ok I can't do this, it is not a DOM element replacement :D

**Comment by arichiardi** [comment]
I will solve too this with #81

**Comment by arichiardi** [comment]
This is solved by 5c8b6d50abae5d087f97fe18ff723378180b4cd2
